src/EditProvider.ts

```diff
@@ -99,16 +100,84 @@ export class EditProvider
   }
 
   public static beautifyOptions(
-    filePath: string = vscode.workspace.rootPath
+    path: string = vscode.workspace.rootPath
   ): Promise<LanguageOptionValues> {
     try {
-      const explorer = cosmiconfig("unibeautify", {});
+      /**
+       * Use a different `.unibeautify` config when vscode config is set and
+       * define if the project config should extend the vscode config.
+       *
+       * @argument {unibeautify.defaultConfig} string path to a file
+       * @argument {unibeautify.extendConfig} boolean extend or not
+       */
+      const vscodeSettings = vscode.workspace.getConfiguration("unibeautify");
+      let defaultConfigOverride = vscodeSettings.defaultConfig;
+      let extendConfig = vscodeSettings.extendConfig;
+      let workspace = path;
+
+      const explorerOptions: any = {};
+      if (defaultConfigOverride) {
+        path = defaultConfigOverride;
+      }
+
+      const explorer = cosmiconfig("unibeautify", explorerOptions);
       const defaultConfig: LanguageOptionValues = {};
-      return explorer
-        .search(filePath)
-        .then(result => (result ? result.config : defaultConfig));
+
+      let isDirectory = fs.lstatSync(path).isDirectory();
+      if (!isDirectory) {
+        return explorer
+          .load(path)
+          .then(returnResult)
+          .catch(catchError);
+      } else {
+        return explorer
+          .search(path)
+          .then(returnResult)
+          .catch(catchError);
+      }
+
+      function returnResult(result: any) {
+        if (!result) return defaultConfig;
+
+        if (defaultConfigOverride && extendConfig) {
+          let resultProject = fs.lstatSync(workspace).isDirectory()
+            ? explorer.searchSync(workspace)
+            : explorer.loadSync(workspace);
+          const config = EditProvider.extend(
+            {},
+            result.config,
+            resultProject.config
+          );
+
+          return config;
+        } else {
+          return result.config;
+        }
+      }
+
+      function catchError() {
+        return vscode.window.showErrorMessage(
+          `We could not find your configuration file: ${path}. Please correct your path, otherwise the plugin will not work!`
+        );
+      }
     } catch (error) {
       return Promise.reject(error);
     }
   }
+
+  public static extend(...obj: any[]) {
```

Comment:
This is going to need unit tests, if it does not already have some.

I would prefer we use https://www.npmjs.com/package/lodash.merge since it is tested and used by many already.

Reply:
Yeah I‘ll add this when we decided to use `extendConfig`, for now I‘ll remove it completely